### PR TITLE
Add dynamic header/footer config determination and update struct

### LINKED ISSUE #11668: Generated Lovelace card config seen as invalid
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The automatically generated Lovelace configuration for an Entities Card with a script (in the footer) and another entity is seen as invalid and cannot be edited (both visually and with YAML) except by the 'Raw configuration editor'.

### Describe the behavior you expected

The automatically generated Lovelace configuration should be considered valid and it should be possible to edit, preview and save the configuration in the Lovelace editor. Or the generated configuration should be changed to be accepted by the validator.

At a minimum, editing the YAML should work. Editing in the visual editor would be nice.

### Steps to reproduce the issue

1. Create a new Lovelace card. In the 'By Entity' tab, select a script and another non-script entity. 
step1
2. Accept the suggestion with the script as a button in the footer.
step2
3. Click on Edit for the newly added card. This will open a YAML editor for the card with the error message:
  > Configuration errors detected:
  > * Required key 'footer.image' is missing.
  > * Required key 'footer.entity' is missing.

step3
  Note that the newly added card was working fine despite these errors.

4. When editing the YAML configuration, loading the preview will fail and the Save button will be disabled.
step4

### What version of Home Assistant Core has the issue?

core-2022.2.6

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Firefox 96.0 (64-bits)

### Which operating system are you using to run this browser?

Kubuntu 18.04

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

```yaml
type: entities
entities:
  - binary_sensor.updater
footer:
  type: buttons
  entities:
    - entity: script.some_script
      show_icon: true
      show_name: true
```

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Alternative / addition to https://github.com/home-assistant/frontend/pull/12792. Ensures we correctly and dynamically determine which config variant applies based on the "type" (similarly to what I added a while back to the entities card config).

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #11668
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/header-footer/structs.ts [hunk 1]
```diff
@@ -22,8 +23,22 @@ export const graphHeaderFooterConfigStruct = object({
...
+      }
+      case "picture": {
+        return pictureHeaderFooterConfigStruct;
+      }
+    }
+  }
+
+  // No "type" property => we fallback to one random variant, which ensure that user gets informed
+  // about missing "type", as all variants have that marked as required.
+  return pictureHeaderFooterConfigStruct;
```
Comment (bramkragten) [inline src/panels/lovelace/header-footer/structs.ts:43]: Why not fallback to all of them?

Comment (bramkragten) [reply src/panels/lovelace/header-footer/structs.ts:43]: union([buttonsHeaderFooterConfigStruct, graphHeaderFooterConfigStruct, pictureHeaderFooterConfigStruct]);

Comment (bramkragten) [inline src/panels/lovelace/header-footer/structs.ts:43]:
```suggestion
  return union([buttonsHeaderFooterConfigStruct, graphHeaderFooterConfigStruct, pictureHeaderFooterConfigStruct]);
```

File: src/panels/lovelace/editor/structs/entities-struct.ts [hunk 2]
```diff
@@ -14,6 +14,8 @@ export const entitiesConfigStruct = union([
     tap_action: optional(actionConfigStruct),
     hold_action: optional(actionConfigStruct),
     double_tap_action: optional(actionConfigStruct),
+    show_icon: optional(boolean()),
+    show_name: optional(boolean()),
```
Comment (bramkragten) [inline src/panels/lovelace/editor/structs/entities-struct.ts:18]: do all entity configs support this?

Comment (spacegaier) [reply src/panels/lovelace/editor/structs/entities-struct.ts]: I took that over from Zack here (https://github.com/home-assistant/frontend/pull/12792), but I don't think all support it yet. In the documentation we mention it for button cards, button rows in entities card, glance card and badges.

So I changed it to a more specific new one derived from `buttonsEntitiesRowConfigStruct`.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**